api/ui_handler.go

```diff
@@ -22,6 +25,12 @@ func NewUIHandler(authService auth.Service) http.Handler {
 
 func NewHandlerWithDefault(root http.FileSystem, handler http.Handler, defaultPath string) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if s3auth := r.Header.Get(sig.V4authHeaderName); strings.HasPrefix(s3auth, sig.V4authHeaderPrefix) {
```

Comment:
For sigv4 this'll work, however, I suggest also checking `s3auth` against `sig.V2AuthHeaderRegexp` to cover all bases.

Reply:
Done.